plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineStore.java

```diff
@@ -266,8 +275,41 @@ private Client client() {
     @FunctionalInterface
     interface ProcessorFactoryProvider {
 
-        Processor.Factory get(Environment environment);
+        Processor.Factory get(Environment environment, TemplateService templateService);
+
+    }
+
+    static class InternalTemplateService implements TemplateService {
```

Comment:
I am starting to wonder if this additional code belongs to PipelineStore. seems like it's become much more than just a store.